lib/AppInfo/Application.php

```diff
@@ -101,6 +101,8 @@ public function registerFilesActivity() {
 		Util::connectHook('\OCA\Files_Trashbin\Trashbin', 'post_restore', 'OCA\Activity\FilesHooksStatic', 'fileRestore');
 		Util::connectHook('OCP\Share', 'post_shared', 'OCA\Activity\FilesHooksStatic', 'share');
 		Util::connectHook('OCP\Share', 'pre_unshare', 'OCA\Activity\FilesHooksStatic', 'unShare');
+		Util::connectHook('\OC\Files\Storage\Shared', 'fopen', 'OCA\Activity\FilesHooksStatic', 'downloadedShare');
+		Util::connectHook('\OC\Files\Storage\Shared', 'file_get_contents', 'OCA\Activity\FilesHooksStatic', 'downloadedShare');
```

Comment:
Yeah this seems to be the only way I can see this working atm is indeed a storage wrapper. The problem is we don't have the file id available at that point so this is a bit taffer